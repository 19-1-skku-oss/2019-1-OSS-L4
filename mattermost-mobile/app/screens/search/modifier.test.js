// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import Preferences from 'mattermost-redux/constants/preferences';

import Modifier from './modifier';

describe('Search RecentItem', () => {
    const item = {
        value: 'in:',
        modifier: 'channel-name',
        description: 'to find posts in specific channels',
    };

    const baseProps = {
        item,
        setModifierValue: jest.fn(),
        theme: Preferences.THEMES.default,
    };

    test('should match snapshot and respond to events', () => {
        const wrapper = shallow(
            <Modifier {...baseProps}/>
        );

        expect(wrapper.getElement()).toMatchSnapshot();
        wrapper.find('TouchableHighlight').first().props().onPress();
        expect(baseProps.setModifierValue).toHaveBeenCalledTimes(1);
        expect(baseProps.setModifierValue).toHaveBeenCalledWith(item.value);
    });
});
