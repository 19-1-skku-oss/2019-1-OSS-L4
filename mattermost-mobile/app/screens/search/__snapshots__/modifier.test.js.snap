// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search RecentItem should match snapshot and respond to events 1`] = `
<TouchableHighlight
  activeOpacity={0.85}
  delayPressOut={100}
  onPress={[Function]}
  underlayColor="rgba(69,120,191,0.5)"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "height": 58,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "paddingHorizontal": 16,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "marginRight": 5,
            }
          }
        >
          <Icon
            allowFontScaling={false}
            name="plus-square-o"
            size={12}
            style={
              Object {
                "color": "rgba(61,60,64,0.5)",
                "fontSize": 16,
              }
            }
          />
        </View>
        <Text
          style={
            Object {
              "color": "#3d3c40",
              "fontSize": 14,
            }
          }
        >
          channel-name
        </Text>
      </View>
      <Text
        style={
          Object {
            "color": "rgba(61,60,64,0.5)",
            "fontSize": 12,
            "marginTop": 5,
          }
        }
      >
        to find posts in specific channels
      </Text>
    </View>
  </View>
</TouchableHighlight>
`;
